webdriverProject/src/18-1dynamicload.spec.ts

```diff
@@ -0,0 +1,67 @@
+// Task 1.
+
+// Разработать тест со следующими шагами:
+
+//   - открыть https://the-internet.herokuapp.com/
+//   - перейти на страницу Dynamic Loading
+//   - Дождаться появления каждой ссылки на странице (их 2)
+//   - кликнуть по ссылке Example 1: Element on page that is hidden
+//   - дождаться появления кнопки start
+//   - кликнуть по кнопке start
+//   - дождаться появления текста "Hello World!" в теге h4 с помощью метода waitForElementWithText(), который вам надо разработать!:)
+
+//  Создать функцию waitForElementWithText(selector, text, timeout) для ожидания определенного текста (text) 
+//  у элемента с определенным селектором (selector) на протяжении определенного времени (timeout):
+//   - Использовать browser.waitUntil с комбинацией проверок (элемент виден и тест верный)
+//   - Добавить понятный timeoutMsg, с пояснением какие проверки не пройдены и селектором элемента
+
+describe("Dynamic loading", () =>{
+    const task1Url = 'https://the-internet.herokuapp.com/';
+    const dynamicLoadingLink = '//*[@href="/dynamic_loading"]';
+    const linksXpath = '//a[contains(text(), "Example")]';
+    const link1Xpath = '//a[contains(text(), "Example 1")]';
+    const link2Xpath = '//a[contains(text(), "Example 2")]';
+
+    async function waitForElementWithText(selector: string, text:string, time:number){
+       await browser.waitUntil(
+        async() =>{
+            const element = await $(selector);
+            const displayed = await element.isDisplayed();
+            const textPresence = await element.getText();
+            return (displayed && textPresence === text);
+        },
+        {
+            timeout: time,
+            timeoutMsg: `The element with text ${text} was not displayed and did not have expected text after ${time} ms waiting`
+        }
+       );
+    }
+    before (async () => {
+        await browser.maximizeWindow();
+        await browser.pause(2000);
+    });
+    
+    beforeEach( async () => {
+    await browser.url(task1Url);
+    await $(dynamicLoadingLink).click();
+    await $(linksXpath).waitForDisplayed();
+    });
+  
+    it('click on example 1', async () => {
```

Comment:
Try to use the same (similar) test naming convention (if possible)
`Should display text "Hello world!" after loading`